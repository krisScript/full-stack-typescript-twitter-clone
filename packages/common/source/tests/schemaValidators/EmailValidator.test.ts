import EmailValidator from '@schemaValidators/EmailValidator';

describe('EmailValidator', (): void => {
  const email = 'testmail@mail.com';
  it(`should validate successfully`, async (): Promise<void> => {
    const body = {
      email,
    };
    await expect(
      EmailValidator.validate(body, { abortEarly: false }),
    ).resolves.toBe(body);
  });
  it(`should throw an error`, async (): Promise<void> => {
    const body = {
      email: '',
    };
    await expect(
      EmailValidator.validate(body, { abortEarly: false }),
    ).rejects.toMatchSnapshot();
  });
});
