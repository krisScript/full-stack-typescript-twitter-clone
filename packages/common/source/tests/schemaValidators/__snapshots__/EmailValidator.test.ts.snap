// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailValidator should throw an error 1`] = `[ValidationError: email is a required field]`;
